common/src/main/java/io/pravega/common/util/ConfigurationOptionsExtractor.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.util;
+
+import lombok.NonNull;
+
+public class ConfigurationOptionsExtractor {
```

Comment:
Javadoc for this class and all of its public methods.

Reply:
Done.